### ISSUE TITLE
[Bug]: Visual focus indicators (VFIs) aren't visible in high contrast mode

### ISSUE THREAD

**Issue by amarhefka-artech** [storybookjs/storybook#25935]
### Describe the bug

Several of the elements in the Storybook UI use a CSS `box-shadow` to render their focus state (the state that's shown when you tab to the item with a keyboard). These box-shadows do not appear in High Contrast Mode—or what are called "Contrast themes" in modern Windows system settings—so there's no way for a keyboard user to tell when they receive focus.

![sc-vfi](https://github.com/storybookjs/storybook/assets/148581341/35f768e5-4ee5-478f-97cc-dab284730935)
![sc-no-vfi](https://github.com/storybookjs/storybook/assets/148581341/a9da3738-9d86-466a-b118-a6a6476722ed)

### To Reproduce

Windows 11 / Chrome
1. Navigate through the interactive elements in the Storybook UI using a keyboard; press the tab key and confirm there's a visible focus indicator for each element that receives focus.
2. Enable a "Contrast theme" in Windows settings. 
3. Repeat step 1. Do the VFIs consistently appear, or are some of them missing?

**Comment by onyxolu** [comment]
New to storybook, will pick this up as my first issue. Details looks clear and straightforward

**Comment by Tomsxmart** [comment]
Removing outline on focused elements might not be the best idea for focus states since when windows high contrast mode  is on browsers will override CSS properties and ignore box-shadows 

A potential fix would be adding the following CSS properties on focus

`    outline-color: transparent;
    outline-width: 1px;
    outline-style: solid;
`

This would insure highlighting on high contrast across browsers. I'm unable to do the PR since storybook does not support windows development (WSL did not work either).

**Comment by StefanFeser** [comment]
When testing our Storybook (v.8.0.6) in High Contrast mode (`force-colors: active`) it also seems like some controls in the args could need some style improvements. Mainly the textareas, switch toggles & selects:

<img width="935" alt="image" src="https://github.com/storybookjs/storybook/assets/455342/d0dd486e-8b1c-4b7b-b3fd-c34b5b550a7e">

**Comment by kawaljen** [comment]
Hello can i work on this one ?
